tests/unit/test_hw_tools.py

```diff
@@ -926,16 +931,87 @@ def test_disk_not_available(self, mock_lshw):
                     "vendor": "NVIDIA Corporation",
                 },
             ],
+            False,
             {HWTool.DCGM},
         ),
+        (
+            [
+                {
+                    "id": "display",
+                    "class": "display",
+                    "handle": "PCI:0000:01:00.0",
+                    "description": "VGA compatible controller",
+                    "product": "GA107M [GeForce RTX 3050 Mobile]",
+                    "vendor": "NVIDIA Corporation",
+                },
+            ],
+            True,
+            set(),
+        ),
     ],
 )
+@mock.patch("hw_tools._is_nvidia_module_blacklisted")
 @mock.patch("hw_tools.lshw")
-def test_nvidia_gpu_verifier(mock_lshw, lshw_output, expect):
+def test_nvidia_gpu_verifier(
+    mock_lshw, mock_is_nvidia_blacklisted, lshw_output, blacklist_output, expect
+):
     mock_lshw.return_value = lshw_output
+    mock_is_nvidia_blacklisted.return_value = blacklist_output
     assert nvidia_gpu_verifier() == expect
 
 
+@pytest.mark.parametrize("modprobe_bool", [True, False])
+@pytest.mark.parametrize("cmdline_bool", [True, False])
```

Comment:
A single parametrize makes easier to test all scenarios
```suggestion
@pytest.mark.parametrize("modprobe_bool, cmdline_bool, excpected", [(True, True, True), (True, False, True), (False, False, True)])
```

Reply:
yes, but flattening would mean that I have to calculate all the combinations myself (and in fact you're missing `(False, True, True)` ;))

Reply:
I support Gabriel's suggestion.

The key here is to declare the possible variables and expected return values explicitly. Otherwise, it results in a redundant implementation within the unit test.

Reply:
I think Andrea's implementation is better and simpler since it's a Cartesian product of two variables. At the same time, I think James concern is also valid, line 972 is exactly the same as the code, and it just feels weird that we are some asserting the expected result

Reply:
given the lack of agreement and blocking requests for changes I'll keep this as is